src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -103,13 +103,13 @@ def generate_pex_arg_list(self) -> Iterator[str]:
 
 
 @rule
-async def digest_complete_platforms(
-    complete_platforms: PexCompletePlatformsField,
+async def digest_complete_platform_addresses(
```

Comment:
I threw up my hands with rule graph errors, unions and the like. This refactor works but its a bit clunky in that the subclassing backends each need to add a ~throwaway rule to link in.

Let me know if there is a better way to do all this. The basic chain is:
1. I want to subclass a field to change its doc (PexCompletePlatformsField -> {PythonAwsLambdaCompletePlatforms, GoogleCloudFunctionCompletePlatforms})
2. There is a pre-existing rule which turns a PexCompletePlatformsField into another thing and the engine does not allow subclasses; so ...